ts/prime/Web/src/routes/index.js

```diff
@@ -6,7 +6,7 @@ import Public from './public'
 
 const CustomRoutes = (props) => {
   const auth = useAuthContext()
-  return auth.loggedIn ? <Authed /> : <Public history={props.history} />
+  return auth.loggedIn ? <Authed kycStatus={auth.kycStatus} /> : <Public history={props.history} />
```

Comment:
Why not switch case here?

```suggestion
switch(auth.status) {
  case AuthStatus.notLoggedIn:
  	return <PublicRoutes>
    break;
  case AuthStatus.loggedIn:
  	return <AuthedRoutes>
    break;
  case AuthStatus.KYCApproved:
  	return <KYCRoutes>
}
```

And then make sure accessing \<AuthedRoutes> or \<KYCRoutes> as notLoggedIn redirects to signup. And accessing \<KYCRoutes> as loggedIn redirects to the submit KYC info page.

Reply:
Since currently we're keep `loggedIn` status and `kycStatus` separately (because some other actions also need to use `kycStatus`), then I guess we will want to switch `kycStatus`.

Reply:
Sure that could work as well. Just make sure we only branch of once and don't have to use a bunch of if else deeper in the code